services/slam/builtin/builtin.go

```diff
@@ -530,9 +492,36 @@ func NewBuiltIn(ctx context.Context, deps registry.Dependencies, config config.S
 		return nil, errors.Wrap(err, "configuring camera error")
 	}
 
-	slamMode, err := RuntimeConfigValidation(svcConfig, string(config.Model.Name), logger)
-	if err != nil {
-		return nil, errors.Wrap(err, "runtime slam config error")
+	modelName := string(config.Model.Name)
+	slamLib, ok := slam.SLAMLibraries[modelName]
+	if !ok {
+		return nil, errors.Errorf("%v algorithm specified not in implemented list", modelName)
+	}
+
+	slamMode, ok := slamLib.SlamMode[svcConfig.ConfigParams["mode"]]
+	if !ok {
+		return nil, errors.Errorf("getting data with specified algorithm %v, and desired mode %v",
+			modelName, svcConfig.ConfigParams["mode"])
+	}
+
+	slamConfig.SetupDirectories(svcConfig.DataDirectory, logger)
+
+	if slamMode == slam.Rgbd || slamMode == slam.Mono {
```

Comment:
[opt] I commented in the other PR that I think this section should go in SetupDirectories.

Reply:
I advice against it, I am not sure we will be able to use `slamMode` as it is right now, since it'll most likely disappear.